### ISSUE TITLE
Header registration for X5T does not allow decoding

### ISSUE THREAD

**Issue by BrianSipos** [TimothyClaeys/pycose#37]
I'm attempting to decode a COSE message with an "x5t" parameter, and what happens when attempting to decode is (only showing the top of the stack):
```
  File "PATH/pycose/cose/cosebase.py", line 19, in from_cose_obj
    uhdr = cls._parse_header(cose_obj.pop(0))
  File "PATH/pycose/cose/cosebase.py", line 93, in _parse_header
    new_hdr[attr.headers.CoseHeaderKeys(k)] = parse_func(v)
TypeError: __init__() missing 1 required positional argument: 'certificate'
```
The registration in `CoseHeaderKeys` uses the class constructor, which doesn't actually decode the header encoded value. Also, the current X5T class does not actually store the thumbprint value so there's no way to decode an actual header value.

It may make better sense for the `X5T` to actually contain the thumbprint bytestring with some static functions to separately: decode a header value and compute a thumbprint (as the current constructor does).

**Comment by BSipos-RKF** [comment]
I made a local change with a similar API to `CoseKey` in https://github.com/BSipos-RKF/pycose/commit/1ddf3a615c7b9ed247434055deb38be6788c4f57

**Comment by BrianSipos** [comment]
This was actually fixed with #46 changes.
